CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSV.java

```diff
@@ -176,117 +215,328 @@ public void setupCSV(ExtensionWebSocketClient oClient ,String fileFolderPath ,St
+
+                chackedAttribute = "content";
+                List<Map<String, Object>> content = (List<Map<String, Object>>) body.get("content");
+
+                for (int i = 0; i < content.size(); i++) {
+
+                    String line = (String) content.get(i).get("text");
+                    bw.write(line);//.getBytes("Cp862")
+                    bw.newLine();
+                }
+
+                bw.close();
+                rsArray = CreateResponse("io.vantiq.csvsource.success", "File Created Succeesfully", file.toString());
+
+            }
+            return rsArray;
+        } catch (InvalidPathException exp) {
+            throw new VantiqCSVException(String.format("Path %s not exist", pathStr, exp));
+        } catch (Exception ex) {
+            throw new VantiqCSVException(
+                    String.format("Illegal request structure , attribute %s doesn't exists", chackedAttribute));
+        }
+    }
+
+    /**
+     * The method used to execute an append command to an existing file , triggered by a SELECT on
+     * the respective source from VANTIQ.
+     * 
+     * @param message
+     * @return
+     * @throws VantiqCSVException
+     */
+    public HashMap[] processAppend(ExtensionServiceMessage message) throws VantiqCSVException {
+        HashMap[] rsArray = null;
+        String chackedAttribute = "Body";
+        String pathStr = "";
+        String fileStr = "";
+
+        try {
+            Map<String, ?> request = (Map<String, ?>) message.getObject();
+            Map<String, Object> body = (Map<String, Object>) request.get("body");
+            chackedAttribute = "path";
+            pathStr = (String) body.get("path");
+            Path path = Paths.get(pathStr);
+            if (!path.toFile().exists()) {
+                rsArray = CreateResponse("io.vantiq.csvsource.nofolder", "Folder Not Exists", pathStr);
+            }
+
+            chackedAttribute = "file";
+            fileStr = (String) body.get("file");
+            String fullFilePath = path.toString() + "\\" + fileStr;
+            File file = new File(fullFilePath);
+            if (file.exists()) {
+                FileWriter fw = new FileWriter(file, true);
+
+                // FileOutputStream fos = new FileOutputStream(file);
+
+                BufferedWriter bw = new BufferedWriter(fw);
+
+                chackedAttribute = "content";
+                List<Map<String, Object>> content = (List<Map<String, Object>>) body.get("content");
+
+                for (int i = 0; i < content.size(); i++) {
+
+                    String line = (String) content.get(0).get("text");
+                    bw.append(line);
+                    bw.newLine();
+                }
+
+                bw.close();
+                fw.close();
+                // fos.close();
+                rsArray = CreateResponse("io.vantiq.csvsource.success", "File Appended Succeesfully", file.toString());
```

Comment:
As noted originally, these should be constants.  You don't want to be retyping the codes each time.

```suggestion
                rsArray = CreateResponse(CSV_SUCCESS, "File Appended Succeesfully", file.toString());
```

Where somewhere else there's a

```
public static String CSV_CODE_BASE = "io.vantiq.extsrc.csvsource.";
public static String CSV_SUCCESS = CSV_CODE_BASE + "success";
public static String CSV_NOFILE = CSV_CODE_BASE + "nofile";
public static String CSV_NOFOLDER = CSV_CODE_BASE + "nofolder";
public static String CSV_FILEEXISTS = CSV_CODE_BASE + "file exists";
```